### PR TITLE
*: fix golangci-lint warnings

### PR DESCRIPTION
**Pull request by sohankunkerkar**
Fixes part of #1121 
Addresses the following warnings:
https://gist.github.com/sohankunkerkar/e7a8a97be039283d841db527acd7d360

And a few more `errcheck` warnings which are not covered in this gist, but appeared in an ad-hoc manner while trying to fix these warnings.

### CODE REVIEW

File: internal/exec/stages/disks/partitions.go [hunk 1]
```diff
@@ -320,9 +320,12 @@ func (p PartitionList) Swap(i, j int) {
 func (s stage) partitionDisk(dev types.Disk, devAlias string) error {
 	if dev.WipeTable != nil && *dev.WipeTable {
 		op := sgdisk.Begin(s.Logger, devAlias)
+		//nolint:errcheck
 		s.Logger.Info("wiping partition table requested on %q", devAlias)
 		op.WipeTable(true)
-		op.Commit()
+		if err := op.Commit(); err != nil {
```
Comment (cgwalters) [inline internal/exec/stages/disks/partitions.go:326]: This is the kind of fix I'd split out separately, missing an error check on a `Commit()` call seems like an important fix.

File: internal/exec/stages/files/files_test.go [hunk 2]
```diff
@@ -29,10 +28,6 @@ func (pw pathWrapper) getPath() string {
 	return string(pw)
 }
 
-func (pw pathWrapper) create(l *log.Logger, u util.Util) error {
```
Comment (bgilbert) [inline internal/exec/stages/files/files_test.go:32]: This is an odd function to accidentally become unused.  It turns out it's trying to fulfill the `filesystemEntry` interface, which changed in fd6103a5d7aef884377748b611596be4b5893d44.  We should figure out what this test is trying to accomplish and fix it.

File: internal/exec/engine.go [hunk 3]
```diff
@@ -91,6 +91,7 @@ func (e Engine) Run(stageName string) error {
 	systemBaseConfig, r, err := system.FetchBaseConfig(e.Logger, e.PlatformConfig.Name())
 	e.logReport(r)
 	if err != nil && err != providers.ErrNoProvider {
+		//nolint:errcheck
```
Comment (bgilbert) [inline internal/exec/engine.go:94]: Hmm, let's discuss this one OOB.

File: tests/filesystem.go [hunk 4]
```diff
@@ -224,6 +224,7 @@ func setupDisk(ctx context.Context, disk *types.Disk, diskIndex int, imageSize i
 	loopdev := disk.Device
 	defer func() {
 		if err != nil {
+			//nolint:errcheck
 			destroyDevice(loopdev)
```
Comment (bgilbert) [inline tests/filesystem.go:228]: Does something like `_ = destroyDevice(loopdev)` satisfy the linter?  That might be cleaner.

File: internal/exec/stages/disks/luks.go [hunk 5]
```diff
@@ -110,7 +110,9 @@ func (s *stage) createLuks(config types.Config) error {
 		var ignitionCreatedKeyFile bool
 		// create keyfile inside of tmpfs, it will be copied to the
 		// sysroot by the files stage
-		os.MkdirAll(distro.LuksInitramfsKeyFilePath(), 0700)
+		if err := os.MkdirAll(distro.LuksInitramfsKeyFilePath(), 0700); err != nil {
+			return err
```
Comment (bgilbert) [inline internal/exec/stages/disks/luks.go:114]: The surrounding code wraps errors before returning them, so we should do that here too.

File: internal/exec/util/passwd.go [hunk 6]
```diff
@@ -218,11 +218,13 @@ func (u Util) AuthorizeSSHKeys(c types.PasswdUser) error {
...
 			return fmt.Errorf("failed to set SSH key: %v", err)
 		}
-		journal.Send(fmt.Sprintf("wrote ssh authorized keys file for user: %s", c.Name), journal.PriInfo, map[string]string{
+		if err := journal.Send(fmt.Sprintf("wrote ssh authorized keys file for user: %s", c.Name), journal.PriInfo, map[string]string{
 			"IGNITION_USER_NAME": c.Name,
 			"IGNITION_PATH":      path,
 			"MESSAGE_ID":         ignitionSSHAuthorizedkeysMessageID,
-		})
+		}); err != nil {
+			return err
```
Comment (bgilbert) [inline internal/exec/util/passwd.go:226]: Likewise, should be wrapped.  Though I'm not sure we should consider it a fatal error if we can't log a journal message.

File: internal/exec/stages/files/files_test.go (hunk 2 above)
Comment (sohankunkerkar) [reply internal/exec/stages/files/files_test.go:32]:
```
internal/exec/stages/files/files_test.go:32:23: func `pathWrapper.create` is unused (unused)
```
Yeah, I agree with you. I was under the impression that this function is just returning a `nil`, and the linter caught it as an unused entry so thought of dropping it. I'm still confused with the `unused` vs `deadcode` warnings.

`unused: Checks Go code for unused constants, variables, functions and types`
`deadcode: Finds unused code`

File: tests/filesystem.go (hunk 4 above)
Comment (sohankunkerkar) [reply tests/filesystem.go:228]: Yeah, the linter doesn't complain after this change.

File: internal/exec/util/passwd.go (hunk 6 above)
Comment (sohankunkerkar) [reply internal/exec/util/passwd.go:226]: How about this one:
```
		_ = journal.Send(fmt.Sprintf("wrote ssh authorized keys file for user: %s", c.Name), journal.PriInfo, map[string]string{
			"IGNITION_USER_NAME": c.Name,
			"IGNITION_PATH":      path,
			"MESSAGE_ID":         ignitionSSHAuthorizedkeysMessageID,
		})
```

File: internal/exec/stages/files/files_test.go [hunk 7]
```diff
@@ -19,56 +19,208 @@ import (
...
+						Path: "/a/../a/b",
+					},
+				},
+			}},
 		},
 	}
 
 	for i, test := range tests {
-		dirs := make([]pathWrapper, len(test.in.data))
+		entries := []pathWrapper{}
```
Comment (arithx) [inline internal/exec/stages/files/files_test.go:215]: Can we not drop `pathWrapper` (and it's interface definition) in favor of just using `filesystemEntry`?

File: internal/providers/vmware/vmware_amd64.go [hunk 8]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/coreos/vcontext/report"
 	"github.com/vmware/vmw-guestinfo/rpcvmx"
 	"github.com/vmware/vmw-guestinfo/vmcheck"
-	"github.com/vmware/vmw-ovflib"
+	ovf "github.com/vmware/vmw-ovflib"
```
Comment (arithx) [inline internal/providers/vmware/vmware_amd64.go:29]: Is this in the wrong commit?

File: tests/filesystem.go [hunk 9]
```diff
@@ -224,7 +224,7 @@ func setupDisk(ctx context.Context, disk *types.Disk, diskIndex int, imageSize i
 	loopdev := disk.Device
 	defer func() {
 		if err != nil {
-			destroyDevice(loopdev)
+			_ = destroyDevice(loopdev)
```
Comment (arithx) [inline tests/filesystem.go:227]: All of the changes in the `tests` package (and it's sub-packages) don't seem related to logging and should probably be moved to their own commit.

File: internal/exec/stages/files/files_test.go (hunk 7 above)
Comment (sohankunkerkar) [reply internal/exec/stages/files/files_test.go]: Yeah, makes sense! Addressed this in the latesh push.

File: internal/providers/vmware/vmware_amd64.go (hunk 8 above)
Comment (sohankunkerkar) [reply internal/providers/vmware/vmware_amd64.go]: That's due to my vscode auto-formatting (Not sure why this change was even suggested). Fixed it.

File: internal/exec/util/user_group_lookup_test.go [hunk 10]
```diff
@@ -63,7 +63,10 @@ func TestUserLookup(t *testing.T) {
 
 	// perform a user lookup to ensure libnss_files.so is loaded
 	// note this assumes /etc/nsswitch.conf invokes files.
-	user.Lookup("root")
+	_, err := user.Lookup("root")
+	if err != nil {
+		t.Fatalf("user lookup failed: %v", err)
```
Comment (arithx) [inline internal/exec/util/user_group_lookup_test.go:68]: Hmm, I wonder if we actually want to be failing here. I wonder if we should maybe update the error message to directly point out that's what we're checking for?

**Review by arithx — APPROVED** [review]

File: internal/exec/stages/files/files_test.go [hunk 11]
```diff
@@ -19,56 +19,193 @@ import (
...
+				},
+			}},
 		},
 	}
 
 	for i, test := range tests {
-		dirs := make([]pathWrapper, len(test.in.data))
+		entries := []filesystemEntry{}
+		dirs := make([]types.Directory, len(test.in.data))
 		for j := range dirs {
```
Comment (bgilbert) [inline internal/exec/stages/files/files_test.go:202]: Let's do `for _, entry := range test.in.data` here, and delete `dirs`.

**Review by bgilbert — COMMENTED** [review]
LGTM generally.

**Review by bgilbert — APPROVED** [review]
